elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -0,0 +1,219 @@
+     * to be used by underlying Elide.get method
+     * @param query query from the Async request
+     * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
+     * @return MultivaluedMap with query parameters
+     */
+    protected MultivaluedMap<String, String> getQueryParams(String query) throws URISyntaxException {
+        URIBuilder uri;
+        uri = new URIBuilder(query);
+        MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>();
+        for (NameValuePair queryParam : uri.getQueryParams()) {
+            queryParams.add(queryParam.getName(), queryParam.getValue());
+        }
+        log.debug("QueryParams: {}", queryParams);
+        return queryParams;
+    }
+
+    /**
+     * This method parses the url and gets the query params and retrieves path
+     * to be used by underlying Elide.get method
+     * @param query query from the Async request
+     * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
+     * @return Path extracted from URI
+     */
+    protected String getPath(String query) throws URISyntaxException {
+        URIBuilder uri;
+        uri = new URIBuilder(query);
+        log.debug("Retrieving path from query");
+        return uri.getPath();
+    }
+
+    /**
+     * This method updates the model for AsyncQuery with passed status value.
+     * @param status new status based on the enum QueryStatus
+     * @param asyncQueryId queryId from asyncQuery request
+     * @throws IOException IOException from DataStoreTransaction
+     * @return AsyncQuery Object
+     */
+    protected AsyncQuery updateAsyncQueryStatus(QueryStatus status, UUID asyncQueryId) throws IOException {
+        log.debug("Updating AsyncQuery status to {}", status);
+        DataStoreTransaction tx = elide.getDataStore().beginTransaction();
+
+        // Creating new RequestScope for Datastore transaction
+        RequestScope scope = new RequestScope(null, null, tx, null, null, elide.getElideSettings());
+
+        EntityProjection asyncQueryCollection = EntityProjection.builder()
+            .type(AsyncQuery.class)
+            .build();
+        AsyncQuery query = (AsyncQuery) tx.loadObject(asyncQueryCollection, asyncQueryId, scope);
+        query.setQueryStatus(status);
+        tx.save(query, scope);
+        tx.commit(scope);
+        tx.flush(scope);
+        tx.close();
+        return query;
+    }
+
+    /**
+     * This method updates the model for AsyncQuery with result object,
+     * @param asyncQueryResult AsyncQueryResult object to be associated with the AsyncQuery object
+     * @param asyncQueryId UUID of the AsyncQuery to be associated with the AsyncQueryResult object
+     * @throws IOException IOException from DataStoreTransaction
+     */
+    protected void updateAsyncQueryStatus(AsyncQueryResult asyncQueryResult, UUID asyncQueryId) throws IOException {
+        log.debug("Updating AsyncQueryResult to {}", asyncQueryResult);
```

Comment:
Right now we have a overloaded method for updateAsyncQueryStatus - one of them is used to update Query Status and the other one is used to add the AsyncQueryResult Object. We do it this way because at any instance we only need to update either the query status or add the result object.  Could you please elaborate on how we can create this generic updateAsyncQuery function ?